src/client/service/VoteFuncationalityHandler.java

```diff
@@ -0,0 +1,93 @@
+package client.service;
+
+import java.io.IOException;
+
+import client.requestgateway.VoteRequestGateway;
+import client.util.InputHandler;
+import client.util.PrintOutToConsole;
+import client.util.RequestHandler;
+import client.util.UserDecisionHandler;
+
+public class VoteFuncationalityHandler {
+    private final String role;
+    private final String userID;
+    private final VoteRequestGateway voteRequestGateway;
+    private final RequestHandler requestHandler;
+    private final UserDecisionHandler userDecisionHandler;
+
+    public VoteFuncationalityHandler(String role, String userID) {
+        this.role = role;
+        this.userID = userID;
+        this.voteRequestGateway = new VoteRequestGateway(role);
+        this.requestHandler = new RequestHandler();
+        this.userDecisionHandler = new UserDecisionHandler();
+    }
+
+    public VoteFuncationalityHandler(String role) {
+        this(role, null);
+    }
+
+    public void viewChefRollout() {
+        String jsonRequest = voteRequestGateway.createChefRolloutRequest();
+        String jsonResponse = requestHandler.sendRequestToServer(jsonRequest);
+        PrintOutToConsole.printToConsole(jsonResponse);
+    }
+
+    public void viewRecommendation() {
+        String jsonRequest = voteRequestGateway.createRecommendationRequest();
+        String jsonResponse = requestHandler.sendRequestToServer(jsonRequest);
+        PrintOutToConsole.printToConsole(jsonResponse);
+    }
+
+    public void addVote() {
```

Comment:
Multiple things are being done including taking user inputs for the voting. Please segregate into different parts.